ipaplatform/redhat/authconfig.py

```diff
@@ -18,15 +18,136 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import logging
+import six
+import abc
+import re
+
 from ipaplatform.paths import paths
 from ipapython import ipautil
 from ipapython.admintool import ScriptError
 import os
 
 FILES_TO_NOT_BACKUP = ['passwd', 'group', 'shadow', 'gshadow']
 
+logger = logging.getLogger(__name__)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class RedHatAuthTool(object):
+
+    @staticmethod
+    def get_tool():
+        return RedHatAuthSelect()
+
+    @abc.abstractmethod
+    def configure(self, sssd, mkhomedir, statestore):
+        pass
+
+    @abc.abstractmethod
+    def unconfigure(self, fstore, statestore,
+                    was_sssd_installed,
+                    was_sssd_configured):
+        pass
+
+    @abc.abstractmethod
+    def backup(self, path):
+        pass
+
+    @abc.abstractmethod
+    def restore(self, path):
+        pass
+
+
+class RedHatAuthSelect(RedHatAuthTool):
+
+    def __get_authselect_current_output(self):
+        current = ipautil.run([
+                paths.AUTHSELECT,
+                "current"
+        ])
+        return current.raw_output
+
+    def __parse_authselect_output(self, output_text=None):
+        if output_text is None:
+            output_text = self.__get_authselect_current_output()
+
+        cfg_params = re.findall(
+            ".*?:\s*(\S+)\s*\\n.*?:\s*(.*)",
+            output_text,
+            re.DOTALL
+        )
+
```

Comment:
It is difficult to understand the regular expression here, and it will be difficult to maintain. Please use @tiran suggestion and call ipautil.run with an env variable in order to force LC_ALL=C.UTF8. This way, you can use a human-friendly regexp with english words.

Reply:
done.